redis-cluster/base/init-job.yaml

```diff
@@ -0,0 +1,89 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: redis-init-scripts
+data:
+  cluster-init.sh: |
+    #!/bin/bash
+    if [[ ! -z "$REDIS_PASSWORD" ]]; then
+      export REDISCLI_AUTH=$REDIS_PASSWORD
+    fi
+    
+    sleep 5
+            
+    for pod_id in $(seq 0  $(($NODE_REPLICA_COUNT-1))); do
+      node="$REDIS_NODE_NAME-$pod_id"   
+    
+      ping_command="timeout 5 redis-cli -h $node.$REDIS_HEADLESS_SERVICE -p $REDIS_PORT ping"     
+      while [[ $($ping_command) != 'PONG' ]]; do
+        echo "Waiting for node $node.$REDIS_HEADLESS_SERVICE to be ready"
+        sleep 5
+      done
+    
+      nodes="$nodes $node.$REDIS_HEADLESS_SERVICE:$REDIS_PORT"
+    done
+    
+    check_output=$(redis-cli --cluster check $REDIS_NODE_NAME-0.$REDIS_HEADLESS_SERVICE:$REDIS_PORT)
+    if [[ ! $check_output =~ "[OK] All 16384 slots covered" ]]; then
```

Comment:
what's so special about the number `16384`? Is it tied to any configuration/will people need to know to update this if they change something else?

Reply:
"The cluster's key space is split into 16384 slots, effectively setting an upper limit for the cluster size of 16384 master nodes (however, the suggested max size of nodes is on the order of ~ 1000 nodes)."

Its a hard coded value in redis clusters. 
More info here https://redis.io/docs/latest/operate/oss_and_stack/reference/cluster-spec/

Reply:
> "The cluster's key space is split into 16384 slots, effectively setting an upper limit for the cluster size of 16384 master nodes (however, the suggested max size of nodes is on the order of ~ 1000 nodes)."
> 
> Its a hard coded value in redis clusters. More info here https://redis.io/docs/latest/operate/oss_and_stack/reference/cluster-spec/

so are we actually interested in that number, or do we just want to know if the command responded ok? If it's the latter case, relying on the exit code seems reasonable (and avoids us relying on the formatting of log messages, which I assume don't have any stability guarantees)

Reply:
The command returns okay in the case where the node is reachable and the cluster is not healthy, for instance if one node is up and the rest are down and the 16384 slots are not all available.

Reply:
> The command returns okay in the case where the node is reachable and the cluster is not healthy, for instance if one node is up and the rest are down and the 16384 slots are not all available.

A more robust alternative might be to check the output of `CLUSTER INFO` https://redis.io/docs/latest/commands/cluster-info, specifically the bit `cluster_state`:

> cluster_state: State is ok if the node is able to receive queries. fail if there is at least one hash slot which is unbound (no node associated), in error state (node serving it is flagged with FAIL flag), or if the majority of masters can't be reached by this node.

Reply:
Its still going to be parsing command output but yea I can change it to that.

Reply:
> Its still going to be parsing command output but yea I can change it to that.

I would favour parsing output that is documented/part of the API over less structured logging output